TinkerPop tests failing due to Vertex equality check

Three TinkerPop tests are currently failing on `master`:

```java
[ERROR] Tests run: 929, Failures: 2, Errors: 1, Skipped: 264, Time elapsed: 145.955 s <<< FAILURE! - in org.janusgraph.blueprints.InMemoryJanusGraphStructureTest
[ERROR] shouldRemoveEdges  Time elapsed: 0.269 s  <<< FAILURE!
java.lang.AssertionError: expected:<199> but was:<200>
	at org.apache.tinkerpop.gremlin.structure.GraphTest.shouldRemoveEdges(GraphTest.java:895)
[ERROR] shouldCommitPropertyAutoTransactionByDefault  Time elapsed: 0.216 s  <<< ERROR!
java.lang.IllegalStateException: The property does not exist as the key has no associated value for the provided element: e[t8-3c0-t1-3c0][4320-l->4320]:name
	at org.apache.tinkerpop.gremlin.structure.TransactionTest.shouldCommitPropertyAutoTransactionByDefault(TransactionTest.java:319)
[ERROR] shouldRollbackPropertyAutoTransactionByDefault  Time elapsed: 0.117 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[yyy]> but was:<[xxx]>
	at org.apache.tinkerpop.gremlin.structure.TransactionTest.shouldRollbackPropertyAutoTransactionByDefault(TransactionTest.java:360)
```

This can be reproduced with:

```bash
mvn clean install -pl janusgraph-test -am -Dtest.skip.tp=false -DskipTests=true
```

I used [`git bisect`](https://git-scm.com/docs/git-bisect) to find the commit that introduced these failures and it was [this one](https://github.com/JanusGraph/janusgraph/commit/27e2c906439234f955a0df6276a647c18c0631fd).

The problematic change was the change of [this line](https://github.com/JanusGraph/janusgraph/pull/1118/files#diff-2cf9bf0eb13b598adc3706e714067159L43) from:

```java
if (v == v.it())
```

to:

```java
if (v.it().equals(v)) {
```

If that line is changed back, then the tests pass again.

--- a/.travis.yml
+++ b/.travis.yml
@@ -48,7 +48,7 @@ env:
     - MODULE='es' ARGS='-Pelasticsearch2'
     - MODULE='es' ARGS='-Pelasticsearch1'
     - MODULE='berkeleyje'
-    - MODULE='test'
+    - MODULE='test' ARGS='-Dtest.skip.tp=false'
     - MODULE='cassandra' ARGS='-Dtest=**/diskstorage/cassandra/thrift/* -Dtest.skip.unordered=true -Dtest.skip.ssl=true -Dtest.skip.serial=true'
     - MODULE='cassandra' ARGS='-Dtest=**/diskstorage/cassandra/thrift/* -Dtest.skip.ordered=true -Dtest.skip.ssl=true -Dtest.skip.serial=true'
     - MODULE='cassandra' ARGS='-Dtest=**/diskstorage/cassandra/thrift/* -Dtest.skip.unordered=true -Dtest.skip.ordered=true'

--- a/janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/AbstractTypedRelation.java
+++ b/janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/AbstractTypedRelation.java
@@ -43,8 +43,7 @@ public abstract class AbstractTypedRelation extends AbstractElement implements I
 
     @Override
     public InternalRelation it() {
-        InternalVertex v = getVertex(0);
-        if (v.it().equals(v)) {
+        if (isLoadedInThisTx()) {
             return this;
         }
 
@@ -54,6 +53,11 @@ public abstract class AbstractTypedRelation extends AbstractElement implements I
         }
 
         return next;
+    }
+
+    private boolean isLoadedInThisTx() {
+        InternalVertex v = getVertex(0);
+        return v == v.it();
     }
 
     @Override

--- a/janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/CacheVertexProperty.java
+++ b/janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/CacheVertexProperty.java
@@ -50,7 +50,7 @@ public class CacheVertexProperty extends AbstractVertexProperty {
 
         if (startVertex.hasAddedRelations() && startVertex.hasRemovedRelations()) {
             //Test whether this relation has been replaced
-            final long id = super.longId();
+            final long id = longId();
             it = Iterables.getOnlyElement(startVertex.getAddedRelations(
                 internalRelation -> (internalRelation instanceof StandardVertexProperty) && ((StandardVertexProperty) internalRelation).getPreviousID() == id), null);
         }
@@ -67,21 +67,15 @@ public class CacheVertexProperty extends AbstractVertexProperty {
     }
 
     private synchronized InternalRelation update() {
-        StandardVertexProperty copy = new StandardVertexProperty(super.longId(), propertyKey(), getVertex(0), value(), ElementLifeCycle.Loaded);
+        StandardVertexProperty copy = new StandardVertexProperty(longId(), propertyKey(), getVertex(0), value(), ElementLifeCycle.Loaded);
         copyProperties(copy);
         copy.remove();
 
         StandardVertexProperty u = (StandardVertexProperty) tx().addProperty(getVertex(0), propertyKey(), value());
-        if (type.getConsistencyModifier()!= ConsistencyModifier.FORK) u.setId(super.longId());
-        u.setPreviousID(super.longId());
+        if (type.getConsistencyModifier()!= ConsistencyModifier.FORK) u.setId(longId());
+        u.setPreviousID(longId());
         copyProperties(u);
         return u;
-    }
-
-    @Override
-    public long longId() {
-        InternalRelation it = it();
-        return (it == this) ? super.longId() : it.longId();
     }
 
     private RelationCache getPropertyMap() {
@@ -120,14 +114,14 @@ public class CacheVertexProperty extends AbstractVertexProperty {
 
     @Override
     public byte getLifeCycle() {
-        if ((getVertex(0).hasRemovedRelations() || getVertex(0).isRemoved()) && tx().isRemovedRelation(super.longId()))
+        if ((getVertex(0).hasRemovedRelations() || getVertex(0).isRemoved()) && tx().isRemovedRelation(longId()))
             return ElementLifeCycle.Removed;
         else return ElementLifeCycle.Loaded;
     }
 
     @Override
     public void remove() {
-        if (!tx().isRemovedRelation(super.longId())) {
+        if (!tx().isRemovedRelation(longId())) {
             tx().removeRelation(this);
         }// else throw InvalidElementException.removedException(this);
     }

--- a/janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java
+++ b/janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java
@@ -2016,6 +2016,19 @@ public abstract class JanusGraphTest extends JanusGraphBaseTest {
     }
 
     @Test
+    public void testPropertyIdAccessInDifferentTransaction() {
+        JanusGraphVertex v1 = graph.addVertex();
+        Object expectedId = v1.property("name", "foo").id();
+        graph.tx().commit();
+
+        VertexProperty p = getOnlyElement(v1.properties("name"));
+
+        // access property id in new transaction
+        graph.tx().commit();
+        assertEquals(expectedId, p.id());
+    }
+
+    @Test
     public void testNestedTransactions() {
         Vertex v1 = graph.addVertex();
         newTx();
@@ -5136,6 +5149,19 @@ public abstract class JanusGraphTest extends JanusGraphBaseTest {
 
         e = Iterables.getOnlyElement(v3.query().direction(Direction.OUT).labels("knows").edges());
         assertEquals(222, e.<Integer>value("uid").intValue());
+    }
+
+    @Test
+    public void testRemoveEdge() {
+        JanusGraphVertex v1 = graph.addVertex();
+        JanusGraphVertex v2 = graph.addVertex();
+        JanusGraphEdge e = v1.addEdge("related", v2);
+        graph.tx().commit();
+
+        e.remove();
+        graph.tx().commit();
+
+        assertFalse(tx.query().edges().iterator().hasNext());
     }
 
    /* ==================================================================================

